Campaign fixtures taxonomy

DFurnes opened the pull request:
## Changes
- Creates an array of fact nodes for the More Facts modal.
- Creates a "Test Issue" issue in the Causes taxonomy if it doesn't already exist.

For review: @aaronschachter @angaither 
## Usage

To create test campaign: `drush campaign-create ../tests/fixtures/campaign.json`
## Screenshot

![test-campaign-taxonomy](https://cloud.githubusercontent.com/assets/583202/3726369/99000870-1692-11e4-968f-bbbb86b0394a.png)

DFurnes committed:
Create array of fact nodes from JSON.

DFurnes committed:
Create issue in causes taxonomy if doesn't already exist.

aaronschachter commented on lib/modules/dosomething/dosomething_taxonomy/dosomething_taxonomy.helpers.inc:
```diff
@@ -0,0 +1,63 @@
+ *
+ * @return Term ID of the found term or else FALSE
+ */
+function dosomething_taxonomy_get_term_from_name($term_name, $vid) {
```
I would rename this as `dosomething_taxonomy_get_tid_by_name` as it returns a tid and not a term object.

aaronschachter commented on lib/modules/dosomething/dosomething_taxonomy/dosomething_taxonomy.helpers.inc:
```diff
@@ -0,0 +1,63 @@
+ * Otherwise, update the existing vocabulary with the new name.
+ * @return vocabulary object
+ */
+function dosomething_taxonomy_create_vocabulary_once($name, $machine_name) {
```
Hmm. I'm not into the naming `create_vocabulary_once` convention.. because by definition a Create operation should just create once.  Plus this doesn't create if one exists. 

The vocabularies should always exist because of the DoSomething Taxonomy features being enabled though.  I think you can just get away with `taxonomy_vocabulary_machine_name_load('issue')`in`dosomething_taxonomy_create_term_once` ...

DFurnes replied:
Yeah, if "Issue" is guaranteed to exist even without an pulling down staging DB then that makes sense.

DFurnes committed:
Remove create_vocabulary method since unnecessary.

Vocabularies should be created already by Features. See PR discussion:
https://github.com/DoSomething/dosomething/pull/2892#discussion-diff-154
87624

DFurnes committed:
Rename get_term to get_tid method for clarity.

DFurnes committed:
Rename create_term_once to create_term.

aaronschachter commented:
:+1:
